safehttp/form.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"errors"
+	"mime/multipart"
+	"strconv"
+)
+
+// Form TODO
+type Form struct {
+	values map[string][]string
+	err    error
+	parsed bool
```

Comment:
Hm... Why do we need this field? Haven't we decided that `req.Form()` and so on will just parse the form, and there is no other way to access the object?

Reply:
We don't actually, I was overly careful but the user can't access the form if it hasn't been parsed in the current design. Thanks for spotting it. Resolved in [1f3a543](url)